custom_components/tahoma/climate_st.py

```diff
@@ -0,0 +1,242 @@
+
+COMMAND_EXIT_DEROGATION = "exitDerogation"
+COMMAND_REFRESH_STATE = "refreshState"
+COMMAND_SET_DEROGATION = "setDerogation"
+COMMAND_SET_MODE_TEMPERATURE = "setModeTemperature"
+
+CORE_DEROGATED_TARGET_TEMPERATURE_STATE = "core:DerogatedTargetTemperatureState"
+CORE_TARGET_TEMPERATURE_STATE = "core:TargetTemperatureState"
+
+PRESET_FREEZE = "Freeze"
+PRESET_NIGHT = "Night"
+
+ST_DEROGATION_TYPE_STATE = "somfythermostat:DerogationTypeState"
+ST_HEATING_MODE_STATE = "somfythermostat:HeatingModeState"
+ST_DEROGATION_HEATING_MODE_STATE = "somfythermostat:DerogationHeatingModeState"
+
+STATE_DEROGATION_FURTHER_NOTICE = "further_notice"
+STATE_DEROGATION_NEXT_MODE = "next_mode"
+STATE_DEROGATION_DATE = "date"
+STATE_PRESET_AT_HOME = "atHomeMode"
+STATE_PRESET_AWAY = "awayMode"
+STATE_PRESET_FREEZE = "freezeMode"
+STATE_PRESET_MANUAL = "manualMode"
+STATE_PRESET_SLEEPING_MODE = "sleepingMode"
+
+MAP_HVAC_MODES = {
+    STATE_DEROGATION_DATE: HVAC_MODE_AUTO,
+    STATE_DEROGATION_NEXT_MODE: HVAC_MODE_HEAT,
+    STATE_DEROGATION_FURTHER_NOTICE: HVAC_MODE_HEAT,
+}
+
+MAP_PRESET_MODES = {
+    STATE_PRESET_AT_HOME: PRESET_HOME,
+    STATE_PRESET_AWAY: PRESET_AWAY,
+    STATE_PRESET_FREEZE: PRESET_FREEZE,
+    STATE_PRESET_MANUAL: PRESET_NONE,
+    STATE_PRESET_SLEEPING_MODE: PRESET_NIGHT,
+}
+PRESET_TEMPERATURES = {
+    PRESET_HOME: "somfythermostat:AtHomeTargetTemperatureState",
+    PRESET_AWAY: "somfythermostat:AwayModeTargetTemperatureState",
+    PRESET_FREEZE: "somfythermostat:FreezeModeTargetTemperatureState",
+    PRESET_NIGHT: "somfythermostat:SleepingModeTargetTemperatureState",
+}
+
+
+class SomfyThermostat(TahomaDevice, ClimateEntity):
+    """Representation of Somfy Smart Thermostat."""
+
+    def __init__(self, device, controller, sensor):
+        """Init method."""
+        super().__init__(device, controller)
+        self._temp_sensor_entity_id = sensor
+        self._saved_target_temp = (
+            self.select_state(PRESET_TEMPERATURES[self.preset_mode])
+            if self.hvac_mode == HVAC_MODE_AUTO
+            else self.select_state(CORE_DEROGATED_TARGET_TEMPERATURE_STATE)
+        )
+        self._current_temperature = None
+
+    async def async_added_to_hass(self):
+        """Register temperature sensor after added to hass."""
+        await super().async_added_to_hass()
+        async_track_state_change(
+            self.hass, self._temp_sensor_entity_id, self._async_temp_sensor_changed
+        )
+
+        @callback
+        def _async_startup(event):
+            """Init on startup."""
+            if self._temp_sensor_entity_id is not None:
+                temp_sensor_state = self.hass.states.get(self._temp_sensor_entity_id)
+                if temp_sensor_state and temp_sensor_state.state != STATE_UNKNOWN:
+                    self.update_temp(temp_sensor_state)
+
+        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, _async_startup)
+
+        self.schedule_update_ha_state(True)
```

Comment:
It probably work, but I don't get it. What's the purpose?

Reply:
line 104? It's to first a first update at startup to get the temperature from the sensor freshly registered. I guess we can get rid of that line.

Reply:
The whole method :)

Reply:
it attaches the temperature sensor to the thermostat entity. It's taken from the generic thermostat where you have to specify the temperature sensor: https://github.com/home-assistant/core/blob/60009ec2f96001fd95713f0a835feac77513d156/homeassistant/components/generic_thermostat/climate.py#L180